vllm/v1/engine/output_processor.py

```diff
@@ -201,18 +201,37 @@ def abort_requests(
         self,
         request_ids: Iterable[str],
     ) -> list[str]:
-        request_ids_to_abort = []
-        for request_id in request_ids:
-            req_state = self.request_states.pop(request_id, None)
+        request_ids_to_abort = self.flatten_req_to_abort(request_ids)
+        self.handle_abort_reqs(request_ids_to_abort)
+        return request_ids_to_abort
+
+    def flatten_req_to_abort(self, req_ids: Iterable[str]) -> list[str]:
+        ret = []
+        for req_id in req_ids:
+            parent = self.parent_requests.pop(req_id, None)
+            if parent is None:
+                ret.append(req_id)
+            else:
+                ret.extend(parent.child_requests)
+        return ret
+
+    # "Aborted request", meaning the frontend first detects that
```

Comment:
This should be a docstring rather than a comment.

Reply:
done